community/kernel/src/main/java/org/neo4j/kernel/impl/factory/GraphDatabaseFacade.java

```diff
@@ -930,16 +904,19 @@ public GraphPropertiesProxy newGraphPropertiesProxy()
     {
         return new GraphPropertiesProxy( this );
     }
+    private static final long UNINITIALIZED = -2L;
+    private static final long NO_ID = -1L;
 
-    private final class NodeCursorResourceIterator implements ResourceIterator<Node>
+    private final class NodeCursorResourceIterator<CURSOR extends NodeIndexCursor> implements ResourceIterator<Node>
     {
-        private final NodeLabelIndexCursor cursor;
+
+        private final CURSOR cursor;
         private final Statement statement;
         private long next;
         private boolean closed;
         private static final long NOT_INITIALIZED = -2L;
 
-        NodeCursorResourceIterator( NodeLabelIndexCursor cursor, Statement statement )
+        NodeCursorResourceIterator( CURSOR cursor, Statement statement )
```

Comment:
There's a problem in the `close` method of this class. Please make it use `IOUtil.closeAll` (or maybe introduce a `closeBoth` if you don't want to allocate the variadic array) or something like that, to ensure that the cursor is closed, even if `statement.close` throws an exception.

Reply:
Fixed using `IOUtil.closeAll`